hal/api/toolchain.h

```diff
@@ -225,6 +225,21 @@
 #endif
 #endif
 
+/* Define '__RAMFUNC' as an attribute to mark a function as residing in RAM.
+ * Use of __RAMFUNC puts a function in the .data section--i.e. the
```

Comment:
Micro-nit - ".data section" is now wrong. Exact details will depend on the toolchain/linker map. Better to just state that it will place the function into RAM.